### PR TITLE
[IMP] core: DatabaseObject replaces _sql_constraints

### PR DESCRIPTION
**Pull request by kmagusiak**
task-3390431

odoo/odoo#175783

### CODE REVIEW

File: content/developer/reference/backend/orm.rst [hunk 1]
```diff
@@ -510,6 +509,27 @@ behavior is desired:
...
+===========
+
+Similarly to fields, you can declare :attr:`~odoo.models.SQLObject` attributes.
+These offer support to declare
+:attr:`~odoo.models.Constraint`,
+:attr:`~odoo.models.Index` and :attr:`~odoo.models.UniqueIndex`.
+The name of the attribute must begin with `_` to avoid name clashes with field
+names.
+
+Most of these objects support a message argument to customize error handling.
```
Comment (ryv-odoo) [inline content/developer/reference/backend/orm.rst:522]: Unique et contraint support, be more explicit IMO.

File: content/developer/reference/backend/orm.rst [hunk 2]
```diff
@@ -510,6 +509,27 @@ behavior is desired:
...
+It can either be a string and its translation will be provided in the internal
+reflected constraint table.
+Otherwise it can be a function that takes as parameter `env` or `env, diag`
```
Comment (ryv-odoo) [inline content/developer/reference/backend/orm.rst:525]: `env` or `env, diag`, did you change that to only accept the second one ?

File: content/developer/howtos/translations.rst [hunk 3]
```diff
@@ -63,8 +63,6 @@ Odoo automatically exports translatable strings from "data"-type content:
   * if ``selection`` is present and a list (or tuple), it's exported
   * if their ``translate`` attribute is set to ``True``, all of their existing
     values (across all records) are exported
-* help/error messages of :attr:`~odoo.models.Model._constraints` and
-  :attr:`~odoo.models.Model._sql_constraints` are exported
```
Comment (Feyensv) [inline content/developer/howtos/translations.rst:67]: Unless the error messages of constraints are not exported anymore, I believe it'd be better to keep a point about it :thinking:

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: content/developer/reference/backend/orm.rst [hunk 4]
```diff
@@ -510,6 +509,30 @@ behavior is desired:
...
+
+You can customize error messages.
+They can either be strings and their translation will be provided in the internal
+reflected constraint table.
+Otherwise, they can be functions that take `(env, diag)` as parameters
+which respectively denote the environment and psycopg diagnostics.
+
+.. code-block:: python
+
+   .. example::
```
Comment (AntoineVDV) [inline content/developer/reference/backend/orm.rst:529]: My bad, these should be swapped.

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

File: content/developer/howtos/translations.rst (hunk 3 above)
Comment (kmagusiak) [reply content/developer/howtos/translations.rst:67]: I just tried exporting translations for base and I see (random example).
```
#. module: base
#: model:ir.model.constraint,message:base.constraint_ir_model_obj_name_uniq
msgid "Each model must have a unique name."
```

It is true that the export is not done from the code anymore, but the table containing the translations of constraints is exported.

@HydrionBurst Could you confirm?

Comment (kmagusiak) [reply content/developer/howtos/translations.rst:67]: - `_constraints` does not exist (already today) - translating is done in the method implementing `@api.constrains`
- `_sql_constraints` does not exist (this PR) - translating is done using the translated field on the reflection of the constraints (model: ir.model.constraints)

**Comment by kmagusiak** [comment]
@robodoo r+

**Comment by kmagusiak** [comment]
@robodoo r+

**Pull request closed without merging**